onnxruntime/core/providers/cuda/cuda_execution_provider.cc

```diff
@@ -1204,8 +1204,9 @@ CUDAExecutionProvider::GetCapability(const onnxruntime::GraphViewer& graph,
       continue;
 
     const auto& node = *p_node;
-    const auto* cuda_kernel_def = GetKernelRegistry()->TryFindKernel(node, Type());
-    if (cuda_kernel_def == nullptr || !node.GetExecutionProviderType().empty()) {
+    const KernelCreateInfo* cuda_kernel_def = nullptr;
+    if (!node.GetExecutionProviderType().empty() ||
+        !(cuda_kernel_def = GetKernelRegistry()->TryFindKernel(node, Type()))) {
```

Comment:
>cuda_kernel_def = GetKernelRegistry()->TryFindKernel(node, Type() [](start = 10, length = 65)

I remember MSVC may throw warning on assign inside if condition. Maybe just continue if node is already assigned?

if (!node.GetExecutionProviderType().empty())
  continue;

const auto* cuda_kernel_def = ...

Reply:
Looks like MSVC is ok with it.